hoot-services/src/main/java/hoot/services/controllers/ingest/CustomScriptResource.java

```diff
@@ -456,45 +554,77 @@ public Response getDefaultScript(@QueryParam("SCRIPT_PATH") String scriptPath) {
 
-            for (File file : files) {
-                String content = FileUtils.readFileToString(file, "UTF-8");
+        try {
+            if (translationId > -1) {
+                // get the display name because that's the file to delete. folder id is to get path for the file
+                Translations folderMapInfo = getTranslationForUser(user, translationId);
+                String translationName = folderMapInfo.getDisplayName();
+
+                // get full directory path for file being deleted
+                TranslationFolder folder = getTranslationFolderForUser(user, folderMapInfo.getFolderId());
+                String getPath = folder.getPath();
+
+                String translationPath = File.separator + translationName;
+                translationPath = getPath != null ? getPath + translationPath : translationPath;
+                File translationFile = new File(SCRIPT_FOLDER, translationPath + ".js");
+
+                String content = FileUtils.readFileToString(translationFile, "UTF-8");
                 JSONObject oScript = getScriptObject(content);
+                delArr.add((JSONObject) oScript.get("HEADER"));
 
-                if (oScript != null) {
-                    JSONObject header = (JSONObject) oScript.get("HEADER");
-                    if (header.get("NAME").toString().equalsIgnoreCase(scriptName)) {
-                        delArr.add(header);
-                        if (!file.delete()) {
-                            throw new IOException("Error deleting script: " + file.getAbsolutePath());
+                translationFile.delete();
+
+                createQuery().delete(translations)
+                    .where(translations.id.eq(translationId))
+                    .execute();
+            } else {
+
+                List<File> files = getFilesInScriptDir();
+                if (files == null) {
+                    throw new IOException("Script directory does not exist.");
+                }
+
+                for (File file : files) {
+                    String content = FileUtils.readFileToString(file, "UTF-8");
+                    JSONObject oScript = getScriptObject(content);
+
+                    if (oScript != null) {
+                        JSONObject header = (JSONObject) oScript.get("HEADER");
+                        if (header.get("NAME").toString().equalsIgnoreCase(identifier)) {
+                            delArr.add(header);
+                            if (!file.delete()) {
+                                throw new IOException("Error deleting script: " + file.getAbsolutePath());
+                            }
+                            break;
                         }
-                        break;
                     }
                 }
             }
         }
         catch (Exception ex) {
-            String msg = "Error deleting script: " + scriptName;
+            String msg = "Error deleting translation. " + ex.getMessage();
```

Comment:
Can the message just be the exception message to match other error http responses?
i.e. remove the "Error deleting translation."